server/xpub-controller/manuscript.js

```diff
@@ -1,17 +1,195 @@
+      )
+      await fileEntity.updateStatus('CANCELLED')
+      await fileEntity.delete()
+      clearInterval(handle)
+      throw err
+    }
+
+    logger.info(`Manuscript Upload S3::end ${filename} | ${manuscriptId}`)
+
+    let title = ''
+    try {
+      // also send source file to conversion service
+      title = await this.scienceBeamApi.extractSemantics(
+        this.config,
+        fileContents,
+        filename,
+        mimeType,
+      )
+    } catch (error) {
+      let errorMessage = ''
+      if (error.error.code === 'ETIMEDOUT' || error.error.connect === false) {
+        errorMessage = 'Request to science beam timed out'
+      } else {
+        errorMessage = error.message
+      }
+      logger.warn('Manuscript conversion failed', {
+        error: errorMessage,
+        manuscriptId,
+        filename,
+      })
+    }
+
+    // After the length file operations above - now update the manuscript...
+    manuscript = await ManuscriptModel.find(manuscriptId, this.user)
+    const oldFileIndex = manuscript.files.findIndex(
+      element => element.type === 'MANUSCRIPT_SOURCE',
+    )
+
+    logger.info(
+      `Manuscript Upload found index ${oldFileIndex} ${
+        fileEntity.filename
+      } | ${manuscriptId}`,
+    )
+
+    if (oldFileIndex >= 0) {
+      const oldFile = await FileModel.find(manuscript.files[oldFileIndex].id)
+      manuscript.files.splice(oldFileIndex, 1)
+      await oldFile.delete()
+    }
+
+    const pendingFileIndex = manuscript.files.findIndex(
+      element => element.type === 'MANUSCRIPT_SOURCE_PENDING',
+    )
+
+    manuscript.files[pendingFileIndex].type = 'MANUSCRIPT_SOURCE'
+
+    logger.info(`Manuscript Upload Manuscript::save ${title} | ${manuscriptId}`)
+    manuscript.meta.title = title
+    await manuscript.save()
+
+    const sourceList = manuscript.files.filter(
+      f => f.type === 'MANUSCRIPT_SOURCE',
+    )
+    const pendingList = manuscript.files.filter(
+      f => f.type === 'MANUSCRIPT_SOURCE_PENDING',
+    )
+    if (sourceList.length !== 1 || pendingList.length !== 0) {
+      logger.error(`Validation failed ${JSON.stringify(manuscript, null, 4)}`)
+      throw new Error(`Validation Failure on ${manuscript.id}`)
+    }
+
+    logger.info(
+      `Manuscript Upload Manuscript::saved ${
+        manuscript.meta.title
+      } | ${manuscriptId}`,
+    )
+
+    clearInterval(handle)
+    pubsub.publish(`${ON_UPLOAD_PROGRESS}.${userId}`, {
+      uploadProgress: 100,
+    })
+    const actualTime = (Date.now() - startedTime) / 1000
+    logger.info(
+      `Manuscript Upload Time, Actual (${actualTime}) , Predicted (${predictedTime}) | ${manuscriptId}`,
+    )
+
+    return ManuscriptModel.find(manuscriptId, this.user)
   }
 
   async update(data) {
-    const userUuid = await UserModel.getUuidForProfile(this.user)
-    const manuscript = await ManuscriptModel.find(data.id, userUuid)
+    const manuscript = await ManuscriptModel.find(data.id, this.user)
```

Comment:
Why did this have to change? I thought we were just moving `uploadManuscript` into the controller...

If we have to change `this.user` to now be the `userId`, we should rename the variable.

Reply:
@diversemix if you're happy with this, I'll make the change

Reply:
@damnhipster ... let me know what you think!